### PR TITLE
CR-3523: Added tests

### PR DESCRIPTION
**Pull request by Tomerva**

### CODE REVIEW

File: src/cloudrail_runner.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import * as path from 'path';
-import * as fs from 'fs';
+import path from 'path';
+import fs from 'fs';
 import {exec} from 'child_process';
```
Comment (AlinaB5) [inline src/cloudrail_runner.ts:3]:
```suggestion
import { exec } from 'child_process';
```

File: src/commands/scan.ts [hunk 2]
```diff
@@ -1,13 +1,14 @@
-import * as vscode from 'vscode';
+import vscode from 'vscode';
+import path from 'path';
+import simpleGit, {SimpleGitOptions} from 'simple-git';
```
Comment (AlinaB5) [inline src/commands/scan.ts:3]:
```suggestion
import simpleGit, { SimpleGitOptions } from 'simple-git';
```

File: src/commands/scan.ts [hunk 3]
```diff
@@ -70,29 +71,26 @@ export async function scan(diagnostics: vscode.DiagnosticCollection) {
...
-        for (let issueItem of failedRule.issue_items) {
+    for (const failedRule of failedRules) {
+        for (const issueItem of failedRule.issue_items) {
             const iacMetadata = issueItem.violating_entity.iac_resource_metadata;
-            let docPath = path.join(terraformWorkingDirectory, iacMetadata.file_name);
-            let document = await vscode.workspace.openTextDocument(docPath);
-            let foundDiagnostics: vscode.Diagnostic[] = [];
+            const docPath = path.join(terraformWorkingDirectory, iacMetadata.file_name);
+            const document = await vscode.workspace.openTextDocument(docPath);
+            const foundDiagnostics: vscode.Diagnostic[] = [];
```
Comment (AlinaB5) [inline src/commands/scan.ts:85]: please use consistent style <Array<RuleResult>>

File: src/commands/scan.ts [hunk 4]
```diff
@@ -70,29 +71,26 @@ export async function scan(diagnostics: vscode.DiagnosticCollection) {
...
 
-            let existingDiagnostics: readonly vscode.Diagnostic[] | undefined = diagnostics.get(document.uri);
+            const existingDiagnostics: readonly vscode.Diagnostic[] | undefined = diagnostics.get(document.uri);
```
Comment (AlinaB5) [inline src/commands/scan.ts:87]: same here

File: src/commands/scan.ts [hunk 5]
```diff
@@ -185,7 +183,7 @@ async function getTerraformWorkingDirectory(): Promise<string | undefined> {
             return;
         }
 
-        return editorDirectoryPath;
+         return editorDirectoryPath;
```
Comment (AlinaB5) [inline src/commands/scan.ts:186]: looks like unintended space

File: src/commands/version.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
...
+import vscode from 'vscode';
 import { Versioning } from '../tools/versioning';
 
 export function cloudrailVersion(): void {    
-    const versionMsg = `Cloudrail Version: ${Versioning.getCloudrailVersion()}` + '\n' + `Extension Version: ${Versioning.getExtensionVersion()}`;
-    vscode.window.showInformationMessage(versionMsg, {modal:true});
-}
\ No newline at end of file
+    const versionMsg = `Cloudrail Version: ${Versioning.getCloudrailVersion()}\nExtension Version: ${Versioning.getExtensionVersion()}`;
+    vscode.window.showInformationMessage(versionMsg, {modal:true});
```
Comment (AlinaB5) [inline src/commands/version.ts:6]: looks like unintended spac

File: src/test/suite/commands/scan.test.ts [hunk 7]
```diff
@@ -0,0 +1,110 @@
+import vscode from 'vscode';
+import sinon, { stub, restore, match, spy } from 'sinon';
+import { describe, beforeEach, it } from 'mocha';
+import { assert } from 'chai';
+import fs = require('fs');
+import path = require('path');
+import _ from './resources/results.json';
```
Comment (AlinaB5) [inline src/test/suite/commands/scan.test.ts:7]: please give a specific name

File: src/test/suite/tools/configuration.test.ts [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+    afterEach(() => {
+        getConfigurationStub.restore();
+        getApiKeyStub.restore();
+    });
+
+    it('Configuration specified in the settings page', async () => {
+        // Arrange
+        getConfigurationStub.withArgs('cloudrail').returns({
+            get: (_s: string) => {
```
Comment (AlinaB5) [inline src/test/suite/tools/configuration.test.ts:26]: please give a proper name _s

File: src/test/suite/commands/scan.test.ts (hunk 7 above)
Comment (Tomerva) [reply src/test/suite/commands/scan.test.ts:7]: its not actually used in the code, its here just so the json will be "compiled" to the out folder.. you still think i need to give it a more proper name?

File: src/test/suite/tools/configuration.test.ts [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    });
+
+    afterEach(() => {
+        getConfigurationStub.restore();
+        getApiKeyStub.restore();
+    });
+
+    it('Configuration specified in the settings page', async () => {
+        // Arrange
+        getConfigurationStub.withArgs('cloudrail').returns({
```
Comment (alonashkenazi) [inline src/test/suite/tools/configuration.test.ts:25]: fix indentation

File: src/commands/version.ts (hunk 6 above)
Comment (Tomerva) [reply src/commands/version.ts:6]: whats not indented here?

**Review by alonashkenazi — APPROVED** [review]

**Pull request merged**